packages/foam-core/src/note-graph.ts

```diff
@@ -1,121 +1,138 @@
-import { Graph, Edge } from 'graphlib';
-import { Position, Point } from 'unist';
-import GithubSlugger from 'github-slugger';
+import { Graph } from 'graphlib';
 import { EventEmitter } from 'events';
+import { Position, Point, URI, ID } from './types';
+import { uriToSlug, hashURI, computeRelativeUri } from './utils';
 
-type ID = string;
-
-export interface Link {
-  from: ID;
-  to: ID;
+export interface NoteSource {
+  uri: URI;
   text: string;
+  contentStart: Point;
+  end: Point;
+  eol: string;
 }
 
-export interface NoteLink {
-  to: ID;
-  text: string;
+export interface WikiLink {
+  type: 'wikilink';
+  slug: string;
   position: Position;
 }
 
+// at the moment we only model wikilink
+export type NoteLink = WikiLink;
+
 export interface NoteLinkDefinition {
   label: string;
   url: string;
   title?: string;
   position?: Position;
 }
 
-export class Note {
-  public id: ID;
-  public frontmatter: object;
-  public title: string | null;
-  public source: string;
-  public path: string;
-  public start: Point;
-  public end: Point;
-  public eol: string;
-  public links: NoteLink[];
-  public definitions: NoteLinkDefinition[];
-
-  constructor(
-    id: ID,
-    frontmatter: object,
-    title: string | null,
-    links: NoteLink[],
-    definitions: NoteLinkDefinition[],
-    start: Point,
-    end: Point,
-    path: string,
-    source: string,
-    eol: string
-  ) {
-    this.id = id;
-    this.frontmatter = frontmatter;
-    this.title = title;
-    this.source = source;
-    this.path = path;
-    this.links = links;
-    this.definitions = definitions;
-    this.start = start;
-    this.end = end;
-    this.eol = eol;
-  }
+export interface Note {
+  title: string | null;
+  slug: string; // note: this slug is not necessarily unique
+  properties: object;
+  // sections: NoteSection[]
+  // tags: NoteTag[]
+  links: NoteLink[];
+  definitions: NoteLinkDefinition[];
+  source: NoteSource;
 }
 
-export type NoteGraphEventHandler = (e: { note: Note }) => void;
+export type GraphNote = Note & {
```

Comment:
I have actually fixed this to make it clearer (unfortunately the previous comment got lost).

But basically now we have `GraphNote` and `GraphConnection`.
Should be easier to understand than before (`NoteInfo`, `Note` and `Connection`)